gatsby-node.js

```diff
@@ -516,18 +517,24 @@ exports.createSchemaCustomization = ({ actions, schema }) => {
     }
     type paragraph__media_text implements Node {
       field_media_image_size: String
-      field_media_text_desc: BodyField      
-      field_media_text_title: String      
-      field_media_text_links: [FieldLink]    
+      field_media_text_desc: BodyField
+      field_media_text_title: String
+      field_media_text_links: [FieldLink]
       relationships: paragraph__media_textRelationships
     }
     type paragraph__media_textRelationships implements Node {
       field_section_column: taxonomy_term__section_columns @link(from: "field_section_column___NODE")
       field_media_text_media: media__imagemedia__remote_videoUnion @link(from: "field_media_text_media___NODE")
       field_button_section: paragraph__section_buttons @link(from:"field_button_section___NODE")
     }
+    type paragraph__modal_video_widget implements Node {
+      relationships: paragraph__modal_video_widgetRelationships

```

Comment:
Schema needs to be updated to address null errors when no Modal Video paragraph content exists, as I get the following errors when I run your branch against a different Drupal backend:

```
Cannot query field "drupal_id" on type "paragraph__modal_video_widget".
ERROR There was an error in your GraphQL query:
It is recommended to explicitly type your GraphQL schema if you want to use optional fields. This way you don't have to add the mentioned "dummy content". Visit our docs to learn how you can define the schema for "paragraph__modal_video_widget":
If you don't expect "drupal_id" to exist on the type "paragraph__modal_video_widget" it is most likely a typo.
```

Reply:
Sorry about that, `drupal_id` has been added to the schema.

Reply:
Something else just occurred to me - what happens if someone adds more than one modal video via Hero Widgets? (of course they're not supposed to, but we have to account for odd/ill-advised actions from content editors). Can you adjust the multidev so people can only add one Modal Video widget?

Reply:
If the editor adds more than one then more than one appears on the page:

![Screen Shot 2022-05-10 at 2 45 04 PM](https://user-images.githubusercontent.com/311164/167727122-098d86e4-d892-4907-bd09-0dbd6b85b23c.png)

We can limit the total number of Widgets in the Hero Widgets area to one, and since the Modal Video Widget is the only widget type allowed there at the moment, then that could resolve this, for now at least. If you want to allow more widget types in the Hero Widgets area in the future then you may have to remove that limit and rethink this.

Drupal has no way to limit the number of any specific _type_ of widget in a Paragraphs entity reference field, only the total number of paragraphs. Additional work would need to be completed to do that, perhaps involving custom code or additional Drupal modules.

Just let me know what you want to do, thanks.

Reply:
Thanks for the clarification. For now, limit the total number of Widgets in the Hero Widgets area to one.

Reply:
Okay then, it's set to allow only a single widget in the Hero Widgets area. I've exported the configuration on Pantheon. If there's anything I can do just let me know!

Reply:
It looks like you need to update the Gatsby Schema as well, because I just got this error when I tried to build the latest changes:

![image](https://user-images.githubusercontent.com/13698158/167910733-86fb32d9-fa4e-4b90-b7cd-2a31bbc22b3f.png)

Reply:
Drat, changing that setting in Drupal meant it no longer returned an array as the field value. I've updated the schema and program page template to accommodate.